src/pages/workspaces/workspace/analysis/modals/AnalysisDuplicator.ts

```diff
@@ -48,71 +57,64 @@ export const AnalysisDuplicator = ({ destroyOld = false, fromLauncher = false, p
-            () => Ajax().Buckets.analysis(googleProject, bucketName, getFileName(printName), toolLabel).copy(`${newName}.${getExtension(printName)}`, bucketName, true) :
-            () => Ajax().AzureStorage.blob(workspaceId, printName).copy(newName)
+        const duplicate = isGoogleWorkspaceInfo(workspaceInfo) ?
+          () => Ajax().Buckets.analysis(workspaceInfo.googleProject, workspaceInfo.bucketName, getFileName(printName), toolLabel).copy(`${newName}.${getExtension(printName)}`, workspaceInfo.bucketName, true) :
+          () => Ajax().AzureStorage.blob(workspaceInfo.workspaceId, printName).copy(newName)
 
-          if (destroyOld) {
-            await rename()
-          } else {
-            await duplicate()
-          }
+        if (destroyOld) {
+          await rename()
+          // @ts-expect-error
+          await Ajax().Metrics.captureEvent(Events.notebookRename, {
+            oldName: printName,
+            newName,
+            workspaceName: workspaceInfo.name,
+            workspaceNamespace: workspaceInfo.namespace
+          })
+        } else {
+          await duplicate()
+          // @ts-expect-error
+          await Ajax().Metrics.captureEvent(Events.notebookCopy, {
+            oldName: printName,
+            newName,
+            fromWorkspaceNamespace: workspaceInfo.namespace,
+            fromWorkspaceName: workspaceInfo.name,
+            toWorkspaceNamespace: workspaceInfo.namespace,
+            toWorkspaceName: workspaceInfo.name
+          })
+        }
 
-          onSuccess()
-          if (fromLauncher) {
-            Nav.goToPath(analysisLauncherTabName, {
-              namespace, name: workspaceName, analysisName: `${newName}.${getExtension(printName)}`, toolLabel
-            })
-          }
-          if (destroyOld) {
-            //@ts-expect-error
-            Ajax().Metrics.captureEvent(Events.notebookRename, {
-              oldName: printName,
-              newName,
-              workspaceName,
-              workspaceNamespace: namespace
-            })
-          } else {
-            //@ts-expect-error
-            Ajax().Metrics.captureEvent(Events.notebookCopy, {
-              oldName: printName,
-              newName,
-              fromWorkspaceNamespace: namespace,
-              fromWorkspaceName: workspaceName,
-              toWorkspaceNamespace: namespace,
-              toWorkspaceName: workspaceName
-            })
-          }
-        } catch (error) {
-          reportError(`Error ${destroyOld ? 'renaming' : 'copying'} analysis`, error)
+        onSuccess()
+        if (fromLauncher) {
+          Nav.goToPath(analysisLauncherTabName, {
+            namespace: workspaceInfo.namespace, name: workspaceInfo.name, analysisName: `${newName}.${getExtension(printName)}`, toolLabel
+          })
         }
-      }
+      })
     }, [`${destroyOld ? 'Rename' : 'Copy'} Analysis`])
   },
   Utils.cond(
     [processing, () => [centeredSpinner()]],
     () => [
```

Comment:
This could return the Fragment directly instead of wrapping it in an array.

Reply:
It believe this does need to be an array, since the return of this Utils.cond resolve to be the children of the `Modal`.

Reply:
Oh, I see. I think I might have written this as:
```ts
h(Modal, { ... }, [Utils.cond(...)])
```
But no need to change.